File: src/main/java/com/vaadin/demo/component/scroller/ScrollerBoth.java
package com.vaadin.demo.component.scroller;

import com.vaadin.demo.DemoExporter;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.orderedlayout.Scroller;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.server.StreamResource;

@Route("scroller-both")
public class ScrollerBoth extends Div {

  public ScrollerBoth() {
    // tag::snippet[]
    Scroller scroller = new Scroller();
    scroller.setWidthFull();
    scroller.setHeight("300px");

    StreamResource imageResource = new StreamResource("reindeer+.jpg",
            () -> getClass().getResourceAsStream("/images/reindeer.jpg"));

    Image img = new Image(imageResource, "A reindeer walking on a snowy lake shore at dusk");
    scroller.setContent(img);

    add(scroller);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<ScrollerBoth> { // hidden-source-line
  } // hidden-source-line
}

File: src/main/frontend/demo/component/scroller/scroller-both.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/scroller';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import img from '../../../../src/main/resources/images/reindeer.jpg?url';

@customElement('scroller-both')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-scroller class="w-full" style="height: 300px">
        <img src="${img}" alt="A reindeer walking on a snowy lake shore at dusk" />
      </vaadin-scroller>
      <!-- end::snippet[] -->
    `;
  }
}
